### DISCUSSION TITLE
Is it possible to break c++17 exclusivity?

### DISCUSSION THREAD

**Discussion by cyreon** [dbus-cxx/dbus-cxx#108] [Q&A] [answered]
First of all, let me congratulate to you for this D-bus implementation. Very nice job!

It may suit our project very well and we may consider using it for IPC.

Unfortunately we have a constraint; namely C++14 standard is the highest possible we can use.

Is it possible to break this constraint somehow without us upgrading GCC? We surely don't want to miss out.

Thank you in advance for your kind help.

**Comment by rm5248** [comment, marked as answer]
Unfortunately it is not possible to use C++14.  The library makes extensive use of [fold expressions](https://en.cppreference.com/w/cpp/language/fold) to extract data.  

If your compiler supports that feature, it may be possible to enable that feature and have things work in C++14 mode.

**Reply by cyreon** [reply]
Thank you so much for the quick response! We will check on our side what we can do. Let's hope for the best. And good luck with your project and keep up with the good work!
